src/core/geometry/qgscircle.h

```diff
@@ -366,13 +356,221 @@ class CORE_EXPORT QgsCircle : public QgsEllipse
      */
     QDomElement asGml3( QDomDocument &doc, int precision = 17, const QString &ns = "gml", QgsAbstractGeometry::AxisOrder axisOrder = QgsAbstractGeometry::AxisOrder::XY ) const;
 
+    /**
+     * \brief Method used to calculate the number of segments for circle approximation
+     * \ingroup core
+     * \since QGIS 3.44
+     */
+    enum class SegmentCalculationMethod
```

Comment:
Can you move this to Qgis namespace please? (see 3.2 https://github.com/qgis/QGIS-Enhancement-Proposals/blob/master/qep-314-coding-style.md#3-language-features )